### ISSUE TITLE
usePress persists interaction through ui state changes 

### ISSUE THREAD

**Issue by dnaploszek** [adobe/react-spectrum#5282]
### Provide a general summary of the issue here

When using usePress any link that will appear in the same place it will be triggered on android / chrome touch devices.

### 🤔 Expected Behavior?

Link interactions shouldn't get propagated down.

### 😯 Current Behavior

Demo: https://codesandbox.io/s/late-haze-xhyxyj

You need to open the demo in chrome inspect + enable touch interaction (either Desktop (Touch) or mobile will be good).
![image](https://github.com/adobe/react-spectrum/assets/17772241/0b631d23-4b3f-4c26-86be-3b7859b1b644)

Clicking on the button on it's left side (where the link appears) will trigger the link as well. 

### 💁 Possible Solution

Haven't investigated yet, but hoping for some guidance. 

### 🔦 Context

Our android app built in cordova has a view change that has a link just after a button. Once the button interaction triggers the view change the link is also triggered. Caused us massive confusion to the android app users

### 🖥️ Steps to Reproduce

1. Open https://codesandbox.io/s/late-haze-xhyxyj in Chrome
2. Open inspect tab + enable mobile touch interactions
3. Click on the visible button on its left side (where the link will appear).

### Version

React-aria: 3.29.1

### What browsers are you seeing the problem on?

Chrome

### If other, please specify.

Chrome with mobile touch. Android devices

### What operating system are you using?

MacOS. Android

**Comment by snowystinger** [comment]
I suspect you're running into this https://github.com/adobe/react-spectrum/issues/1513 could you confirm?

**Comment by dnaploszek** [comment]
Yeah I suppose it's the same root issue. Crazy how it's been around for 2 years and browsers are still plagued by it 😞
